core/src/main/java/org/jdbi/v3/core/mapper/reflect/ConstructorInstanceFactory.java

```diff
@@ -36,4 +51,36 @@ T newInstance(Object... params) {
     public String toString() {
         return constructor.toString();
     }
+
+    private static <T> boolean isRecordConstructor(Constructor<T> factory)
+    {
+        // If there is more than one constructor, ensure that the factory constructor is the canonical constructor.
+        // If so it will need to get the type parameters from the declared fields.
+        if (factory.getParameters().length != Arrays.stream(factory.getDeclaringClass().getDeclaredFields())
+            .filter(field -> !Modifier.isStatic(field.getModifiers()))
```

Comment:
Should `isStatic` be statically imported?

Reply:
I usually wouldn't, unless it is used many times and the readability would improve.